### PR TITLE
Reduce jitter

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/supergloo/issues/395

### CODE REVIEW

File: pkg/registration/pubsub.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"sync"
+	"time"
+
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/eventloop"
+	"go.uber.org/zap"
+)
+
+type Reciever <-chan EnabledConfigLoops
```
Comment (marcogschmidt) [inline pkg/registration/pubsub.go]: Nit: typo

File: pkg/registration/pubsub.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+type PubSub struct {
+	subscriberCache []chan EnabledConfigLoops
+	subscriberLock  sync.RWMutex
+}
+
+func NewPubsub() *PubSub {
+	return &PubSub{}
+}
+
+func (r *PubSub) Subscribe() Reciever {
```
Comment (marcogschmidt) [inline pkg/registration/pubsub.go]: Looks like `Subscribe`/`Unsubscribe` are only called in this package. If so, can we make them private?

File: pkg/registration/pubsub_test.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+				watches = append(watches, reciever)
+				go func() {
+					<-reciever
+					recievedUpdates++
+				}()
+			}
+			pubSub.publish(ctx, EnabledConfigLoops{})
+			Eventually(func() int {
+				return recievedUpdates
+			}, time.Second*15, time.Second/2).Should(Equal(4))
```
Comment (ilackarms) [inline pkg/registration/pubsub_test.go]: 15 seconds seems a bit long here

Comment (EItanya) [reply pkg/registration/pubsub_test.go]: yea, but it pings every half of a second so unless something goes really wrong, it should get it on the second one most likely

**Pull request merged**